### PR TITLE
feat: SSR - enabling reuseCurrentRendering, specifying new SSR default options, and merging with the ones provided by customers

### PR DESCRIPTION
**Pull request by znikola**
In https://github.com/SAP/spartacus/pull/13680 we introduced an SSR optimization "reuseCurrentRendering" which was hidden behind a config option. In this PR we are enabling it by default.
We are also merging the default SSR options with the ones provided by clients.

closes https://jira.tools.sap/browse/CXSPA-401

### CODE REVIEW

File: core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts [hunk 1]
```diff
@@ -110,3 +110,10 @@ export enum RenderingStrategy {
   DEFAULT = 0,
   ALWAYS_SSR = 1,
 }
+
+export const defaultSsrOptimizationOptions: SsrOptimizationOptions = {
+  concurrency: 20,
+  timeout: 3_000,
+  maxRenderTime: 300_000,
```
Comment (Platonn) [inline core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts:124]: maxRenderTime - is it a new default added by the way? Should we document it in new 6.0 migration docs?

File: core-libs/setup/tsconfig.lib.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "declaration": true,
     "sourceMap": true,
     "inlineSources": true,
+    "strict": true,
```
Comment (Platonn) [inline core-libs/setup/tsconfig.lib.json:11]: 👍

File: core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts (hunk 1 above)
Comment (znikola) [reply core-libs/setup/ssr/optimized-engine/ssr-optimization-options.ts:124]: No, it was set to 300000 ms (5 mins) from the start.
We are just making it more obvious by explicitly setting it here.

**Review by Platonn — DISMISSED** [review]

**Review by WeizhengSap — DISMISSED** [review]

File: core-libs/setup/ssr/engine-decorator/ng-express-engine-decorator.ts [hunk 3]
```diff
@@ -42,10 +45,7 @@ export class NgExpressEngineDecorator {
 
 export function decorateExpressEngine(
   ngExpressEngine: NgExpressEngine,
-  optimizationOptions: SsrOptimizationOptions | null = {
-    concurrency: 20,
-    timeout: 3000,
-  }
+  optimizationOptions: SsrOptimizationOptions | null = defaultSsrOptimizationOptions
```
Comment (Platonn) [inline core-libs/setup/ssr/engine-decorator/ng-express-engine-decorator.ts]: Do we _really_ need to provide a default value here? We're already `{...spreading}` the default options [in OptimizedSsrEngine in line 66](https://github.com/SAP/spartacus/pull/15924/files#diff-1db882c5fb0276ec1884fe195efa449fcc93d591c38b783ca45396de81dbf36fR66)

File: projects/storefrontapp/server.ts [hunk 4]
```diff
@@ -21,9 +22,10 @@ import { AppServerModule } from './src/main.server';
 const express = require('express');
 
 const ssrOptions: SsrOptimizationOptions = {
-  concurrency: 20,
-  timeout: Number(process.env['SSR_TIMEOUT'] ?? 3000),
-  reuseCurrentRendering: true,
+  ...defaultSsrOptimizationOptions,
```
Comment (Platonn) [inline projects/storefrontapp/server.ts]: Do we _really_ need to pass default options here? We're already spreading them in the SsrOptimizedEngine.

Comment (znikola) [reply projects/storefrontapp/server.ts]: nope, removed :)

**Review by WeizhengSap — APPROVED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**